src/jobrunner.rs

```diff
@@ -140,10 +162,26 @@ impl JobRunner {
                 }
             }
 
-            // If there are jobs whose stderr/stdout have closed, keep waiting on them until they
-            // exit.
+            // Iterate over the running jobs and:
+            //   * If any jobs have exceeded their timeout, send them SIGTERM.
+            //   * If there are jobs whose stderr/stdout have closed, keep waiting on them until
+            //     they exit.
             num_waiting = 0;
+            next_finish_by = None;
             for i in 0..self.running.len() {
+                if let Some(Job {
+                    finish_by,
+                    ref child,
+                    ..
+                }) = self.running[i]
+                {
+                    if finish_by <= Instant::now() {
```

Comment:
So does this rely on the event loop waking up?

Would the code still work if we had maxjobs jobs which all hang?

Reply:
This makes `poll` wake up in time to check the next timeout.

Reply:
jolly good :)